lib/plugins/ray_trace.js

```diff
@@ -1,28 +1,35 @@
-const Vec3 = require('vec3')
+const { Vec3 } = require('vec3')
+const { RaycastIterator } = require('../iterators')
 
-module.exports = inject
-
-function inject (bot) {
-  const rayTraceBlock = (maxSteps = 256, vectorLength = 5 / 16) => {
-    const { height, position, yaw, pitch } = bot.entity
-    const cursor = position.offset(0, height, 0)
+module.exports = (bot) => {
+  function getViewDirection (pitch, yaw) {
+    const csPitch = Math.cos(pitch)
+    const snPitch = Math.sin(pitch)
+    const csYaw = Math.cos(yaw)
+    const snYaw = Math.sin(yaw)
+    return new Vec3(-snYaw * csPitch, snPitch, -csYaw * csPitch)
+  }
 
-    const x = -Math.sin(yaw) * Math.cos(pitch)
-    const y = Math.sin(pitch)
-    const z = -Math.cos(yaw) * Math.cos(pitch)
+  bot.blockInSight = (maxSteps = 256, vectorLength = 5 / 16) => {
+    console.warn('[deprecated] use bot.blockAtCursor instead')
+    bot.blockAtCursor(maxSteps * vectorLength)
+  }
 
-    const step = new Vec3(x, y, z).scaled(vectorLength)
+  bot.blockAtCursor = (maxDistance = 256) => {
+    const { position, height, pitch, yaw } = bot.entity
 
-    for (let i = 0; i < maxSteps; ++i) {
-      cursor.add(step)
+    const eyePosition = position.offset(0, height, 0)
+    const viewDirection = getViewDirection(pitch, yaw)
 
-      // TODO: Check boundingBox of block
```

Comment:
I'd preserve this comment. Maybe in future we'll add it

Reply:
done